tests/components/deconz/test_device_trigger.py

```diff
@@ -135,14 +129,294 @@ async def test_get_triggers(hass, aioclient_mock):
+    )
+
+    trigger_config = {
+        CONF_PLATFORM: "device",
+        CONF_DOMAIN: DECONZ_DOMAIN,
+        CONF_DEVICE_ID: device.id,
+        CONF_TYPE: "unsupported",
+        CONF_SUBTYPE: device_trigger.CONF_TURN_ON,
+    }
+
+    assert await async_setup_component(
+        hass,
+        AUTOMATION_DOMAIN,
+        {
+            AUTOMATION_DOMAIN: [
+                {
+                    "trigger": trigger_config,
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {"some": "test_trigger_button_press"},
+                    },
+                },
+            ]
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_entity_ids(AUTOMATION_DOMAIN)) == 0
+
+
+async def test_attach_trigger_no_matching_event(
+    hass, aioclient_mock, mock_deconz_websocket
+):
+    """Test no matching event for device doesn't return a trigger config."""
+    config_entry = await setup_deconz_integration(hass, aioclient_mock)
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get_or_create(
+        config_entry_id=config_entry.entry_id,
+        identifiers={(DECONZ_DOMAIN, "d0:cf:5e:ff:fe:71:a4:3a")},
+        name="Tradfri switch",
+        model="TRADFRI on/off switch",
+    )
+
+    trigger_config = {
+        CONF_PLATFORM: "device",
+        CONF_DOMAIN: DECONZ_DOMAIN,
+        CONF_DEVICE_ID: device.id,
+        CONF_TYPE: device_trigger.CONF_SHORT_PRESS,
+        CONF_SUBTYPE: device_trigger.CONF_TURN_ON,
+    }
+
+    with pytest.raises(InvalidDeviceAutomationConfig):
+        await device_trigger.async_attach_trigger(hass, trigger_config, None, None)
+
+    assert await async_setup_component(
+        hass,
+        AUTOMATION_DOMAIN,
+        {
+            AUTOMATION_DOMAIN: [
+                {
+                    "trigger": trigger_config,
+                    "action": {
+                        "service": "test.automation",
+                        "data_template": {"some": "test_trigger_button_press"},
+                    },
+                },
+            ]
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_entity_ids(AUTOMATION_DOMAIN)) == 1
+
+    # async_attach_trigger raise InvalidDeviceAutomationConfig
+
+    assert not await async_initialize_triggers(
+        hass,
+        [trigger_config],
+        action=Mock(),
+        domain=AUTOMATION_DOMAIN,
+        name="mock-name",
+        log_cb=Mock(),
+    )
+
+
 async def test_helper_no_match(hass, aioclient_mock):
     """Verify trigger helper returns None when no event could be matched."""
     await setup_deconz_integration(hass, aioclient_mock)
-    deconz_event = device_trigger._get_deconz_event_from_device_id(hass, "mock-id")
-    assert not deconz_event
+    assert not device_trigger._get_deconz_event_from_device_id(hass, "mock-id")
```

Comment:
Do we still need these two tests?

Reply:
Coverage wasn't affected so I removed them